app/src/main/java/org/stepic/droid/code/ui/CodeAnalyzer.kt

```diff
@@ -0,0 +1,104 @@
+package org.stepic.droid.code.ui
+
+import org.stepic.droid.util.countWhile
+import org.stepic.droid.util.substringOrNull
+
+/**
+ * Class for smart code analyzing
+ */
+object CodeAnalyzer {
+    private const val LINE_BREAK = '\n'
+
+    private val brackets = hashMapOf(
+        "{" to "}",
+        "(" to ")",
+        "[" to "]"
+    )
+
+    private val indentation = hashMapOf( // code specific indentation symbols
+            "py" to hashSetOf(
```

Comment:
why second argument is hashset, not just a symbol?

Reply:
Because in future we can add more than one symbol for specified languages